tests/sentry/middleware/test_api_gateway.py

```diff
@@ -0,0 +1,98 @@
+from urllib.parse import parse_qs
+
+import requests
+import responses
+from django.conf.urls import url
+from django.test import override_settings
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.testutils import TestCase
+from sentry.types.region import Region, RegionCategory
+from sentry.utils import json
+
+SENTRY_REGION_CONFIG = [
+    Region(name="region1", id=1, address="region1", category=RegionCategory.MULTI_TENANT),
+]
+
+
+@control_silo_endpoint
+class ControlEndpoint(Endpoint):
+    def get(self, request):
+        return Response({"ok": True})
+
+
+urlpatterns = url(r"^/control-endpoint", ControlEndpoint.as_view(), name="control-endpoint")
+
+
+def verify_request_body(body, headers):
+    """Wrapper for a callback function for responses.add_callback"""
+
+    def request_callback(request):
+        assert json.loads(request.body) == body
+        assert (request.headers[key] == headers[key] for key in headers)
+        return 200, {}, b"{}"
+
+    return request_callback
+
+
+def verify_request_params(params, headers):
+    """Wrapper for a callback function for responses.add_callback"""
+
+    def request_callback(request):
+        request_params = parse_qs(request.url.split("?")[1])
+        assert request_params == params
+        assert (request.headers[key] == headers[key] for key in headers)
+        return 200, {}, b"{}"
+
+    return request_callback
+
+
+def verify_file_body(file_body, headers):
+    """Wrapper for a callback function for responses.add_callback"""
+
+    def request_callback(request):
+        assert request.body == file_body
+        assert (request.headers[key] == headers[key] for key in headers)
+        return 200, {}, b"{}"
+
+    return request_callback
+
+
+@override_settings(SENTRY_REGION_CONFIG=SENTRY_REGION_CONFIG, ROOT_URLCONF=__name__)
```

Comment:
The custom `override_regions` decorator might work better than `override_settings` here, to ensure that you don't have to worry about the global region map being cached. Even if this is working in isolation, I don't know offhand whether there's any risk of Pytest running multiple tests in an unpredictable order and polluting the cache unexpectedly in the future.

You can use both together, like
```
@override_regions(SENTRY_REGION_CONFIG)
@override_settings(ROOT_URLCONF=__name__)
```